origin/metrics.go

```diff
@@ -206,120 +208,120 @@ func newMuxerMetrics() *muxerMetrics {
 		},
-		[]string{"connection_id"},
+		append(baseMetricLabels, "connection_id"),
 	)
 	prometheus.MustRegister(maxConcurrentRequestsPerTunnel)
 
 	timerRetries := prometheus.NewGauge(
 		prometheus.GaugeOpts{
-			Name: "timer_retries",
+			Name: "argotunnel_timer_retries",
 			Help: "Unacknowledged heart beats count",
 		})
 	prometheus.MustRegister(timerRetries)
 
-	responseByCode := prometheus.NewCounterVec(
-		prometheus.CounterOpts{
-			Name: "response_by_code",
-			Help: "Count of responses by HTTP status code",
-		},
-		[]string{"status_code"},
-	)
-	prometheus.MustRegister(responseByCode)
+	// responseByCode := prometheus.NewCounterVec(
+	// 	prometheus.CounterOpts{
+	// 		Name: "argotunnel_response_by_code",
+	// 		Help: "Count of responses by HTTP status code",
+	// 	},
+	// 	[]string{"status_code"},
+	// )
+	// prometheus.MustRegister(responseByCode)
 
-	responseCodePerTunnel := prometheus.NewCounterVec(
+	responses := prometheus.NewCounterVec(
 		prometheus.CounterOpts{
-			Name: "response_code_per_tunnel",
-			Help: "Count of responses by HTTP status code fore each tunnel",
+			Name: "argotunnel_responses",
+			Help: "Count of responses for each tunnel",
 		},
-		[]string{"connection_id", "status_code"},
+		append(baseMetricLabels, "connection_id", "status_code"),
 	)
-	prometheus.MustRegister(responseCodePerTunnel)
+	prometheus.MustRegister(responses)
 
 	serverLocations := prometheus.NewGaugeVec(
 		prometheus.GaugeOpts{
-			Name: "server_locations",
+			Name: "argotunnel_server_locations",
 			Help: "Where each tunnel is connected to. 1 means current location, 0 means previous locations.",
 		},
-		[]string{"connection_id", "location"},
+		append(baseMetricLabels, "connection_id", "location"),
 	)
 	prometheus.MustRegister(serverLocations)
 
 	return &TunnelMetrics{
 		haConnections:                  haConnections,
-		totalRequests:                  totalRequests,
-		requestsPerTunnel:              requestsPerTunnel,
+		requests:                       requests,
 		concurrentRequestsPerTunnel:    concurrentRequestsPerTunnel,
-		concurrentRequests:             make(map[string]uint64),
+		concurrentRequests:             make(map[uint64]uint64),
 		maxConcurrentRequestsPerTunnel: maxConcurrentRequestsPerTunnel,
-		maxConcurrentRequests:          make(map[string]uint64),
+		maxConcurrentRequests:          make(map[uint64]uint64),
 		timerRetries:                   timerRetries,
-		responseByCode:                 responseByCode,
-		responseCodePerTunnel:          responseCodePerTunnel,
-		serverLocations:                serverLocations,
-		oldServerLocations:             make(map[string]string),
-		muxerMetrics:                   newMuxerMetrics(),
+
+		responses:          responses,
+		serverLocations:    serverLocations,
+		oldServerLocations: make(map[uint64]string),
+		muxerMetrics:       newMuxerMetrics(baseMetricLabels),
+	}
+}
+
+func hashLabelValues(labelValues []string) uint64 {
```

Comment:
ConnectionID is unique per tunnel. Is there other reason to use the hash value of labels as key?

Reply:
seems to be '0' everywhere, apparently StartTunnelDaemon sets it https://github.com/cloudflare/cloudflared/blob/master/origin/tunnel.go#L135  Bug?

Reply:
ConnectionID is 0 in that case because config.HAConnections is set to 1.